tests/phpunit/CRM/Contribute/Form/Contribution/ConfirmTest.php

```diff
@@ -0,0 +1,151 @@
+    // create pending contribution
+    $contribution = $this->callAPISuccess('Contribution', 'create', array(
+      'contact_id' => $contactID,
+      'financial_type_id' => CRM_Core_PseudoConstant::getKey('CRM_Contribute_BAO_Contribution', 'financial_type_id', 'Campaign Contribution'),
+      'currency' => 'USD',
+      'total_amount' => 100.00,
+      'contribution_status_id' => CRM_Core_PseudoConstant::getKey('CRM_Contribute_BAO_Contribution', 'contribution_status_id', 'Pending'),
+      'contribution_page_id' => $contributionPageID1,
+      'source' => 'backoffice pending contribution',
+    ));
+
+    // create a contribution page which is later used to make online payment for pending contribution
+    $result = $this->callAPISuccess('ContributionPage', 'create', array(
+      'title' => 'Test Contribution Page',
+      'financial_type_id' => CRM_Core_PseudoConstant::getKey('CRM_Contribute_BAO_Contribution', 'financial_type_id', 'Campaign Contribution'),
+      'currency' => 'USD',
+      'financial_account_id' => 1,
+      'payment_processor' => $paymentProcessorID,
+      'is_active' => 1,
+      'is_allow_other_amount' => 1,
+      'min_amount' => 10,
+      'max_amount' => 1000,
+    ));
+    $form = new CRM_Contribute_Form_Contribution_Confirm();
+    $contributionPageID2 = $result['id'];
+    $form->_id = $contributionPageID2;
+    $form->_values = $result['values'][$contributionPageID2];
+    $form->_paymentProcessor = array(
+      'id' => $paymentProcessorID,
+      'billing_mode' => CRM_Core_Payment::BILLING_MODE_FORM,
+      'object' => Civi\Payment\System::singleton()->getById($paymentProcessorID),
+      'is_recur' => FALSE,
+      'payment_instrument_id' => CRM_Core_PseudoConstant::getKey('CRM_Contribute_BAO_Contribution', 'payment_instrument_id', 'Credit card'),
+    );
+    $form->_params = array(
+      'qfKey' => 'donotcare',
+      'contribution_id' => $contribution['id'],
+      'credit_card_number' => 4111111111111111,
+      'cvv2' => 234,
+      'credit_card_exp_date' => array(
+        'M' => 2,
+        'Y' => 2021,
+      ),
+      'credit_card_type' => 'Visa',
+      'email-5' => 'test@test.com',
+      'total_amount' => 100.00,
+      'payment_processor_id' => $paymentProcessorID,
+      'amount' => 100,
+      'tax_amount' => 0.00,
+      'year' => 2021,
+      'month' => 2,
+      'currencyID' => 'USD',
+      'is_pay_later' => 0,
+      'invoiceID' => '6e443672a9bb2198cc12f076aed70e7a',
+      'is_quick_config' => 1,
+      'description' => $contribution['values'][$contribution['id']]['source'],
+      'skipLineItem' => 0,
+    );
+
+    $result = CRM_Contribute_BAO_Contribution_Utils::processConfirm($form,
+      $form->_params,
+      $contactID,
+      $form->_values['financial_type_id'],
+      0, FALSE
+    );
+    // Based on the processed contribution, complete transaction which update the contribution status based on payment result.
+    if (!empty($result['contribution'])) {
```

Comment:
we don't really need to wrap this because it is such a shallow wrapper around  civicrm_api3('contribution', 'completetransaction', array( I think we should ditch the wrapper & just call the api within the test (which is also how 50% of payment processors will call the completion)